### PR TITLE
Refactored config

### PR DESCRIPTION
**Pull request by Drugsosos**
## Description

I'm not fond of the way config was written, so I changed it a bit.
Now it way more readable and accessible.

* Added typing
* Added explanations
* Changed the way `checks` was provided in `handle_input`
* Replaced `crawl_and_check` w/ `nested_get` & `nested_set`

## Issue Fixes

None I guess, I just didn't like the way it was written:
* w/o typing & explanation
* the way `checks` was provided in `handle_input`
* the way config template was looking, it's hard to read at least

<details>
  <summary>Checklist</summary>
  
  </br>

- [x] I am pushing changes to the **develop** branch
- [x] I am using the recommended development environment
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have formatted and linted my code using python-black and pylint
- [x] I have cleaned up unnecessary files
- [x] My changes generate no new warnings
- [x] My changes follow the existing code-style
- [x] My changes are relevant to the project

</details>

## How to test the changes

1. Rename `config.toml` to `config.toml.bak`
2. Run the script
3. Do your tests
4. Delete new `config.toml`
5. Rename `config.toml.bak` to `config.toml`

> Try to play w/ different scenarios, for instance, what if one field is incorrect

### CODE REVIEW

File: utils/settings.py [hunk 1]
```diff
@@ -1,170 +1,260 @@
...
+        name: Name of the value to be checked
 
-    incorrect = False
-    if value == {}:
-        incorrect = True
-    if not incorrect and "type" in checks:
+    Returns:
+        Correct value
+    """
+    correct = True if value else False
```
Comment (Drugsosos) [inline utils/settings.py:55]:
```suggestion
    correct = True if value != {} else False
```

File: utils/settings.py [hunk 2]
```diff
@@ -1,170 +1,260 @@
...
+        obj = obj.setdefault(key, {})
+    obj[keys[-1]] = value
+
 
+def check_vars(
+        path: list,
+        checks: dict,
+) -> None:
+    """
+    Checks if value is in nested dict and correct by path of keys
```
Comment (Drugsosos) [inline utils/settings.py:157]:
```suggestion
    Checks if there is the value in the dict and it's correct 
```

File: utils/settings.py [hunk 3]
```diff
@@ -1,170 +1,260 @@
...
-                )
-                return False
-    except FileNotFoundError:
-        console.print(
-            f"""[blue]Couldn't find {config_file}
-Creating it now."""
-        )
+        # except Exception as error:
+        #     console.print(f"[red bold]Encountered error when trying to to load {kwargs['template_name']}: {error}")
+        #     return False
```
Comment (Drugsosos) [inline utils/settings.py:209]: forgot to uncomment this)
```suggestion
        except Exception as error:
            console.print(f"[red bold]Encountered error when trying to to load {kwargs['template_name']}: {error}")
            return False
```

File: utils/settings.py [hunk 4]
```diff
@@ -1,170 +1,260 @@
...
-    console.print(
-        """\
-[blue bold]###############################
-#                             #
-# Checking TOML configuration #
-#                             #
-###############################
-If you see any prompts, that means that you have unset/incorrectly set variables, please input the correct values.\
-"""
-    )
```
Comment (Drugsosos) [inline utils/settings.py:162]: forgot to revert this

**Review by Drugsosos — COMMENTED** [review]
bruh, fix this later

**Comment by JasonLovesDoggo** [comment]
Actually don't even need that

**Comment by JasonLovesDoggo** [comment]
Dude go through the enhancement list and do that Why fix things that aren't broken...

**Comment by JasonLovesDoggo** [comment]
Besides nobody is ever supposed to actually edit that file except for a dev who can read it fine

**Comment by Drugsosos** [comment]
@JasonLovesDoggo 
You sure you want to close it?
I know it works fine, but it's not great.
* config template looks bad and is in fact unreadable, + not following toml rulebook
* 2 recursive functions is hell to debug if something will ever happen
* typings will never hurt nobody
* it's kinda bad code if the code is only for `dev who can read it fine`, everyone should understand it

Just give it a look. It's not that big, I renamed lots of thing there unintentionally but it's still almost the same code

**Comment by Drugsosos** [comment]
> Dude go through the enhancement list and do that Why fix things that aren't broken...

Yup, I will. While I'm waiting on new-webdriver to be merged I wanted to address thing no-one would

**Pull request closed without merging**